plugins/calibre.koplugin/wireless.lua

```diff
@@ -203,6 +203,11 @@ Do you want to continue? ]]), driver),
 end
 
 function CalibreWireless:connect()
+    if not NetworkMgr:isConnected() then
+        NetworkMgr:beforeWifiAction(function() CalibreWireless:connect() end)
+        return
+    end
```

Comment:
No risk of perpetual recursion if beforeWifiAction fails? (I guess the callback is called just on success so you're 100% sure `NetworkMgr:isConnected() ` is true when the recursive call is done ? haven't checked in your code, cheaper to just ask :)

Reply:
Yes. Yes there is :D.

That was on the list of things to look into, because if the callback actually works, that's indeed a (low, but not null) risk ;).

Reply:
Actually, with the `turn_on` action, the callback will *definitely* only fire if the connection was successful (i.e., `isConnected`).

Things potentially get dicier if the check is actually an `isOnline`.

And `prompt` is possibly also iffier, i'm not sure if the callback is only called on success...

Reply:
There, fixed `prompt` to do the same ;).